Add docs for custom plugins.

Xantier opened the pull request:
Docs for triggering workflow still pending.

Some sentences/words are a bit clunky there. Would appreciate some feedback

Preview: https://60cc8a87df0c4d00085ece7d--roadie.netlify.app/docs/custom-plugins/configuring/

dtuite commented on content/docs/custom-plugins/configuring/index.md:
```diff
@@ -0,0 +1,77 @@
+
+## Introduction
+
+Roadie supports all standard Backstage plugin types that can be built using instructions defined in [Backstage documentation][backstage-plugin-documentation]. Your plugins can be published via npm or yarn, like publishing a normal package. We will provide you functionality to automatically add custom plugins from repositories to your Roadie instance. If you want to have a private hosted repositor ...
```
> Roadie supports all standard Backstage plugin types

We don't support backend plugins yet.

> If you want

I would split out a new paragraph from this point on. The topic changes slightly at this point.

dtuite commented on content/docs/custom-plugins/configuring/index.md:
```diff
@@ -0,0 +1,77 @@
+
+### Step 1. Create plugin definition in Roadie
+
+You can navigate to https://your-company.roadie.so/administration/custom-plugins where you will find a collection of cards detailing custom plugins configured for the Roadie instance. 
```
I wish "https://your-company.roadie.so/administration/custom-plugins" didn't show up as a link in the docs. I think people will click it and be confused when it doesn't work.

dtuite commented on content/docs/custom-plugins/configuring/index.md:
```diff
@@ -0,0 +1,77 @@
+The values that need to be added are as follows:
+
+##### Package name
+Package field corresponds to the name of the NPM package. This can be a public NPM package hosted by Microsoft in npmjs.com or a private package securely hosted in Roadie Artifactory. If the package is hosted in Roadie Artifactory, the name of the package needs to start with `@<your-company>-roadie/`.
```
```suggestion
Package field corresponds to the name of the NPM package. This can be a public NPM package hosted in npmjs.com or a private package securely hosted in Roadie Artifactory. If the package is hosted in Roadie Artifactory, the name of the package needs to start with `@<your-company>-roadie/`.
```

The fact that Microsoft owns npmjs.com is not relevant.

dtuite commented on content/docs/custom-plugins/configuring/index.md:
```diff
@@ -0,0 +1,77 @@
+Package field corresponds to the name of the NPM package. This can be a public NPM package hosted by Microsoft in npmjs.com or a private package securely hosted in Roadie Artifactory. If the package is hosted in Roadie Artifactory, the name of the package needs to start with `@<your-company>-roadie/`.
+
+
+##### Plugin Name
```
```suggestion
##### Plugin name
```

The other section titles are not capitalized.

dtuite commented on content/docs/custom-plugins/configuring/index.md:
```diff
@@ -0,0 +1,77 @@
+
+
+##### Plugin version
+Semantic versioning definition for the plugin to use. This can be defined with standard NPM semantic versioning range identifiers like `^` or `~`, as well as keywords like `latest`.
```
This answers one question I had 👍

dtuite commented on content/docs/custom-plugins/configuring/index.md:
```diff
@@ -0,0 +1,77 @@
+
+
+##### Components
+List of components that the plugin exposes. These are defined as a type and name pair. The type is  one of Card, Content or Page. The name needs to match the component name in code that is built into the plugin package.
```
I would use the word "exports" somewhere in this paragraph since that is the JS/TS keyword which needs to be in place. Is it worth adding a code block as an example of a valid export?

dtuite commented on content/docs/custom-plugins/configuring/index.md:
```diff
@@ -0,0 +1,77 @@
+
+### Step 2. Installing
+
+Custom plugins are installed within the application from an external workflow. This can be triggered either by calling a webhook from your CI pipeline or manually or automatically when a new NPM package is published to private, secure Roadie Artifactory. Note that this installation step may take up to 20 minutes at the moment, though we are tirelessly working on bringing that number down.
```
```suggestion
Custom plugins are installed within the application from an external workflow. This can be triggered in a number of ways:

1. by calling a webhook from your CI pipeline,
2. manually by [doing what???]
3. automatically when a new NPM package is published to private, secure Roadie Artifactory. 
```

I would omit the info about the build pipeline. It's likely to change over time and they don't need to know it in order to learn how to use the feature.

dtuite commented on content/docs/custom-plugins/configuring/index.md:
```diff
@@ -0,0 +1,77 @@
+Newly added plugin configuration. An attempt will be made to install this plugin during the next application installation cycle.
+
+##### Updated
+Updated plugin configuration. An older version of the plugin is still available to be used in the application and updates to version or components will be added during the next application installation cycle
```
```suggestion
Updated plugin configuration. An older version of the plugin is still available to be used in the application and updates to version or components will be added during the next application installation cycle.
```

dtuite commented on content/docs/custom-plugins/configuring/index.md:
```diff
@@ -0,0 +1,77 @@
+Updated plugin configuration. An older version of the plugin is still available to be used in the application and updates to version or components will be added during the next application installation cycle
+
+##### Installed successfully
+Application installation cycle has started and has successfully updated the application code. The plugin and components will be available to use when the next version of the application is rolled out.
```
What is the "application" in this case? The plugin? Or the Backstage application? What is a roll out and how do I know when that will happen?

Xantier commented:
Addressed review comments. New preview: https://deploy-preview-265--roadie.netlify.app/docs/custom-plugins/configuring/

Opted to not add code samples since those are visible in: https://backstage.io/docs/plugins/plugin-development

dtuite commented on content/docs/custom-plugins/artifactory/index.md:
```diff
@@ -11,10 +11,7 @@ your plugins, and we host a dedicated npm registry which you can publish to.
-self-service going forward but currently Roadie doesn't support this out-of-the-box. For now, we will need you to
-fill out [this form][form] the first time you want to add a new plugin. Pushing new versions will not require you
-to fill the form again.
+To be able to use private artifactory and it's automated workflow in your Roadie instance, we need to set it up. You can contact us to set up a repository for you. After the repository is set up, your plugins will be hosted securely on Roadie artifactory and are usable in your Roadie instance.
```
```suggestion
Contact us to get added to our private and secure plugin artefact repository.
```

dtuite commented on content/docs/custom-plugins/configuring/index.md:
```diff
@@ -0,0 +1,82 @@
+
+Roadie supports all standard Backstage frontend plugin types that can be built using instructions defined in [Backstage documentation][backstage-plugin-documentation]. Your plugins can be published via npm or yarn, like publishing a normal package. We will provide you functionality to automatically add custom plugins from repositories to your Roadie instance. 
+
+If you want to have a private hosted repository for your Roadie plugins with automatic update workflow you can follow instructions on [setting up private secure Artifactory for Roadie plugins.](/docs/custom-plugins/artifactory/)
```
```suggestion
If you you would like to use our private hosted repository for your Roadie plugins, complete with automatic update workflow, you can follow instructions on [setting up private secure Artifactory for Roadie plugins.](/docs/custom-plugins/artifactory/)
```

dtuite commented on content/docs/custom-plugins/configuring/index.md:
```diff
@@ -0,0 +1,82 @@
+
+## Prerequisites
+
+Custom plugins need to be enabled for your Roadie instance. You can enable this by contacting us.
```
```suggestion
Contact us to have the custom plugins feature enabled for your Roadie instance.
```

dtuite commented on content/docs/custom-plugins/configuring/index.md:
```diff
@@ -0,0 +1,82 @@
+You can create a new custom plugin definition by clicking "Add New Plugin" floating button. 
+![custom_plugin_form](custom_plugin_form.png)
+
+The values that need to be added are as follows:
```
```suggestion
The required values are as follows:
```

dtuite commented on content/docs/custom-plugins/configuring/index.md:
```diff
@@ -0,0 +1,82 @@
+
+
+##### Plugin name
+The name of the plugin. This field can be arbitrarily chosen and will be used as a identifier when adding components from the plugin to Roadie as Cards, Tabs or Pages.
```
```suggestion
The name of the plugin. This field can be arbitrarily chosen and will be used as an identifier when adding components from the plugin to Roadie as Cards, Tabs or Pages.
```

dtuite commented on content/docs/custom-plugins/configuring/index.md:
```diff
@@ -0,0 +1,82 @@
+Each plugin exposes a status field indicating if they are available to be used within the Roadie instance. The statuses are as follows:
+
+#### Added
+Newly added plugin configuration. An attempt will be made to install this plugin during the next application installation cycle.
```
I'd love a little more info about this application installation cycle. Is it a loop that checks for new plugin versions and installs them? How often does it run?

Xantier replied:
The installation is explained in the step above, is that enough? I think we want to additionally add a documentation page on how to trigger these installations after we have confirmed the full flow

```
### Step 2. Installing

Custom plugins are installed within the application from an external workflow. This can be triggered in a number of ways:
1. By calling a webhook from your CI pipeline,
2. By calling a webhook manually
3. Automatically when a new NPM package is published to private, secure Roadie Artifactory.
```

dtuite replied:
Ah! Sorry. I understand now. The words "application installation cycle" aren't in the section above so I didn't make the connection.

dtuite replied:
```suggestion
Newly added plugin configuration. The new configuration will be installed when the installation workflow is triggered, as described in Step 2 above.
```

dtuite reviewed (changes requested):
I shortened a few paragraphs and added one question.

dtuite commented on content/docs/custom-plugins/configuring/index.md:
```diff
@@ -0,0 +1,82 @@
+### Plugin Statuses
+Each plugin exposes a status field indicating if they are available to be used within the Roadie instance. The statuses are as follows:
+
+#### Added
```
```suggestion
##### Added
```

dtuite commented on content/docs/custom-plugins/configuring/index.md:
```diff
@@ -0,0 +1,82 @@
+Newly added plugin configuration. An attempt will be made to install this plugin during the next application installation cycle.
+
+##### Updated
+Updated plugin configuration. An older version of the plugin is still available to be used in the application and updates to version or components will be added during the next application installation cycle.
```
```suggestion
Updated plugin configuration. An older version of the plugin is still available to be used in your Roadie instance. Updates to versions or components will be added when the installation workflow is triggered, as described in Step 2 above.
```

dtuite commented on content/docs/custom-plugins/configuring/index.md:
```diff
@@ -0,0 +1,82 @@
+Updated plugin configuration. An older version of the plugin is still available to be used in the application and updates to version or components will be added during the next application installation cycle.
+
+##### Installed successfully
+Backstage application installation cycle has started and has successfully updated the application code. The plugin and components will be available to use when the next version of your Roadie instance is rolled out.
```
```suggestion
The installation workflow, as described in Step 2, has been triggered and has successfully updated the application code. The plugin and components will be available to use when the next version of your Roadie instance is rolled out.
```

dtuite reviewed (changes requested):
Ok I think this is the last pass. I think I just need to try this out to really wrap my head around it.

Xantier commented:
Aight, comments addressed. I'll leave this unmerged until we roll the feature out properly

dtuite commented:
Thanks for humouring me 😄
